lib/search.rb

```diff
@@ -1191,6 +1199,10 @@ def posts_eager_loads(query)
       topic_eager_loads << :tags
     end
 
+    Search.custom_topic_eager_loads.each do |block|
+      block.call(topic_eager_loads, search_pms: @search_pms)
```

Comment:
I think we're giving the plugins too much flexibility here by accepting a block? Instead, can the plugin register the exact table names that should be eager loaded?

Reply:
Sorry I just saw the implementation in https://github.com/discourse/discourse-encrypt/pull/21. Instead of allowing the block to mutate `topic_eager_loads`, I think having the block explicitly return an array of table names to be pushed into the `topic_eager_loads` array will be clearer.

Reply:
Also, should we also support plugins being able to pass in a table name(s) directly instead of having to do so through a block?